extractor/image/image.go

```diff
@@ -93,3 +119,33 @@ func fileOpener(fileName string) func() (io.ReadCloser, error) {
 		return ioutil.NopCloser(r), nil
 	}
 }
+
+func tryOCI(fileName string) (v1.Image, error) {
+	lp, err := layout.FromPath(fileName)
+	if err != nil {
+		return nil, xerrors.Errorf("unable to open %s as an OCI Image: %w", fileName, err)
+	}
+
+	index, err := lp.ImageIndex()
+	if err != nil {
+		return nil, xerrors.Errorf("unable to retrieve index.json: %w", err)
+	}
```

Comment:
I'm curious as to when would this error happen? I tried to test it by removing `index.json` from a test image but that case is actually handled by the previous if condition.

Reply:
I'm not sure the case is useful, but I added it. https://github.com/aquasecurity/fanal/pull/101/commits/0e7df0c642f570e1930d83fe32ef7d40477b93a7

An index.json with no permission should also fail at `ImageIndex()`.